hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -140,8 +140,8 @@ void writeGenericRecord(BufferObjectDataOutput output, CompactGenericRecord reco
         Collection<FieldDescriptor> fields = schema.getFields();
         for (FieldDescriptor fieldDescriptor : fields) {
             String fieldName = fieldDescriptor.getFieldName();
-            FieldType fieldType = fieldDescriptor.getType();
-            fieldOperations(fieldType).writeFieldFromRecordToWriter(writer, record, fieldName);
+            FieldKind fieldKind = fieldDescriptor.getKind();
+            fieldOperations(fieldKind).writeFieldFromRecordToWriter(writer, record, fieldName);
```

Comment:
Question: `writeGenericRecord ` method reads from generic record and writes to writer buffer. The generic record keeps a buffer internally, is this buffer different from what will be the buffer for the writer? I was thinking that they are the same and generic record is just another way of accessing the fields. If they are the same, why not bulk copy the bytes?

Reply:
Yes, this is one of the pending improvements that we can apply. There is more refactoring to be done when the buffer is from off-heap to on-heap or vice versa. I left those improvements to a later point to start simple.